File: Easy/tassacsingconf/meta_tassacsingconfdefaultview/meta_tassacsingconfdefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_tassacsingconfdefaultview
{
    public class Meta_tassacsingconfdefaultview : Meta_easydata 
	{
        public Meta_tassacsingconfdefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "tassacsingconfdefaultview") {
				Name = "Definizione dei costi dei corsi singoli";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idtassacsingconf"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "tassacsingconf_aa", "Anno accademico", nPos++);
						DescribeAColumn(T, "tassacsingconf_costomax", "Costo massimo", nPos++);
						if (T.Columns.Contains("tassacsingconf_costomax")) T.Columns["tassacsingconf_costomax"].ExtendedProperties["format"] = "fixed.2";
						DescribeAColumn(T, "costoscontodef_title", "Costo", nPos++);
						DescribeAColumn(T, "costoscontodef_2_title", "Costo corsi speciali", nPos++);
						DescribeAColumn(T, "costoscontodef_sconto_title", "Sconto", nPos++);
						DescribeAColumn(T, "tassacsingconf_numerosconto", "Numero di insegnamenti per cui si applica lo sconto", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/tassacsingconf/meta_tassacsingconfdefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_tassacsingconfdefaultview() {
        MetaData.apply(this, ["tassacsingconfdefaultview"]);
        this.name = 'meta_tassacsingconfdefaultview';
    }

    meta_tassacsingconfdefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_tassacsingconfdefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'aa', 'Anno accademico', null, 1000, 9);
						this.describeAColumn(table, 'tassacsingconf_costomax', 'Costo massimo', 'fixed.2', 3000, null);
						this.describeAColumn(table, 'costoscontodef_title', 'Costo', null, 4200, 2024);
						this.describeAColumn(table, 'costoscontodef2_title', 'Costo corsi speciali', null, 5200, 2024);
						this.describeAColumn(table, 'costoscontodefsconto_title', 'Sconto', null, 6200, 2024);
						this.describeAColumn(table, 'tassacsingconf_numerosconto', 'Numero di insegnamenti per cui si applica lo sconto', null, 7000, null);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idtassacsingconf"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			//$getSorting$

        });

    window.appMeta.addMeta('tassacsingconfdefaultview', new meta_tassacsingconfdefaultview('tassacsingconfdefaultview'));

	}());
